src/cbmc/all_properties.cpp

```diff
@@ -78,20 +78,16 @@ safety_checkert::resultt bmc_all_propertiest::operator()()
   {
     if(it->is_assert())
     {
-      irep_idt property_id;
+      irep_idt property_id = it->get_property_id();
 
-      if(it->source.pc->is_assert())
-        property_id=it->source.pc->source_location.get_property_id();
-      else if(it->source.pc->is_goto())
+      if(property_id.empty())
+        continue;
+
+      if(it->source.pc->is_goto())
       {
         // this is likely an unwinding assertion
```

Comment:
I think this comment no longer is a good fit here. Maybe put it before the `if` and say "goto may yield and unwinding assertion"?